internal/service/project/resource_project.go

```diff
@@ -755,17 +755,16 @@ Else consider the aggregate dependents idle.
 If we get a defined error response, return that right away
 Else retry
 */
-func resourceProjectDependentsDeletingRefreshFunc(ctx context.Context, projectID *string, connV2 *admin.APIClient) retry.StateRefreshFunc {
+func ResourceProjectDependentsDeletingRefreshFunc(ctx context.Context, projectID string, client GroupProjectService) retry.StateRefreshFunc {
 	return func() (any, string, error) {
-		nonNullProjectID := conversion.StringPtrNullIfEmpty(projectID)
-		clusters, _, err := connV2.ClustersApi.ListClusters(ctx, nonNullProjectID.String()).Execute()
+		clusters, _, listClustersErr := client.ListClusters(ctx, projectID)
 		dependents := AtlasProjectDependants{AdvancedClusters: clusters}
 
-		if _, ok := admin.AsError(err); ok {
-			return nil, "", err
+		if _, ok := admin.AsError(listClustersErr); ok {
```

Comment:
(this would imply removing redudant if statement below)
```suggestion
		if listClustersErr != nil {
```
curious if we want to unify both error checks, don't see a reason why for ApiError we would return error, while for others SDK error types keep on retrying. Would simply return error for an encountered error.

Reply:
I didn't want to change this to avoid changing behaviour, but I think it makes sense to unify it.